crates/egui/src/widgets/image.rs

```diff
@@ -43,15 +44,93 @@ use epaint::{util::FloatOrd, RectShape};
 #[must_use = "You should put this widget in an ui with `ui.add(widget);`"]
 #[derive(Debug, Clone)]
 pub struct Image<'a> {
-    source: ImageSource<'a>,
+    id: Id,
+    sources: ImageSources<'a>,
     texture_options: TextureOptions,
     image_options: ImageOptions,
     sense: Sense,
     size: ImageSize,
     pub(crate) show_loading_spinner: Option<bool>,
 }
 
+#[derive(Debug, Clone)]
+pub enum ImageSources<'a> {
+    Multi(Vec<(ImageSource<'a>, Duration)>),
+    Single(ImageSource<'a>),
+}
+
+impl<'a> ImageSources<'a> {
+    pub fn get_source(&self, ctx: &Context, id: Id) -> &'a ImageSource<'_> {
```

Comment:
https://rust-lang.github.io/api-guidelines/naming.html#getter-names-follow-rust-convention-c-getter
```suggestion
    pub fn source(&self, ctx: &Context, id: Id) -> &'a ImageSource<'_> {
```